File: java/variables.java
import java.util.Scanner;

public class Operaciones {
    public static void main(String[] args) {
        // Variables
        int a = 5, b = 2;
        int c;
        
        // Sumar
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: +");
        System.out.println("La segunda variable es: " + b);
        c = a + b;
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName()); // Muestra el tipo de la variable

        // Restar
        a = 3;
        b = 4;
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: -");
        System.out.println("La segunda variable es: " + b);
        c = a - b;
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName());

        // Multiplicar
        a = 8;
        b = 6;
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: *");
        System.out.println("La segunda variable es: " + b);
        c = a * b;
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName());

        // Dividir
        a = 4;
        b = 6;
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: /");
        System.out.println("La segunda variable es: " + b);
        c = a / b;
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName());

        // Exponenciación (potencia)
        a = 15;
        b = 3;
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: **");
        System.out.println("La segunda variable es: " + b);
        c = 1;
        for (int i = 0; i < b; i++) {
            c *= a;
        }
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName());

        // Módulo
        a = 236;
        b = 22;
        System.out.println("La primera variable es: " + a);
        System.out.println("El signo de operacion es: %");
        System.out.println("La segunda variable es: " + b);
        c = a % b;
        System.out.println("El resultado es: " + c);
        System.out.println("Tipo de c: " + ((Object)c).getClass().getName());

        // Operaciones con entrada del usuario
        Scanner scanner = new Scanner(System.in);
        System.out.print("Ingrese el valor para a: ");
        a = scanner.nextInt();
        System.out.print("Ingrese el valor para b: ");
        b = scanner.nextInt();

        int res;

        res = a + b;
        System.out.println("La suma de los numero

File: cpp/variables.cpp
#include <iostream>
#include <typeinfo> // Para obtener el tipo de la variable
using namespace std;

int main()
{

    int a = 5, b = 2;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: +" << endl;
    cout << "La segunda variable es: " << b << endl;
    int c = a + b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl; // Muestra el tipo de la variable

    a = 3;
    b = 4;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: -" << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a - b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    a = 8;
    b = 6;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: *" << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a * b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    a = 4;
    b = 6;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: /" << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a / b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    a = 15;
    b = 3;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: **" << endl;
    cout << "La segunda variable es: " << b << endl;
    c = 1;
    for (int i = 0; i < b; i++)
    {
        c *= a;
    }
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    a = 236;
    b = 22;
    cout << "La primera variable es: " << a << endl;
    cout << "El signo de operacion es: %" << endl;
    cout << "La segunda variable es: " << b << endl;
    c = a % b;
    cout << "El resultado es: " << c << endl;
    cout << "Tipo de c: " << typeid(c).name() << endl;

    // Operaciones con entrada del usuario
    cout << "Ingrese el valor para a: ";
    cin >> a;
    cout << "Ingrese el valor para b: ";
    cin >> b;

    res = a + b;
    cout << "La suma de los numeros es: " << res << endl;

    res = a - b;
    cout << "La resta de los numeros es: " << res << endl;

    res = a * b;
    cout << "La multiplicacion de los numeros es: " << res << endl;

    return 0;
}
